pkg/controller/miganalytic/miganalytics_controller.go

```diff
@@ -0,0 +1,582 @@
+
+			excluded := false
+			compatible := true
+			for _, exclude := range excludedResources {
+				if exclude == r.Name {
+					excluded = true
+				}
+			}
+
+			if len(gv) > 1 {
+				group = gv[0]
+				version = gv[1]
+			} else {
+				version = gv[0]
+			}
+
+			gvr := schema.GroupVersionResource{
+				Group:    group,
+				Version:  version,
+				Resource: r.Name,
+			}
+
+			for _, ns := range incompatible.Namespaces {
+				if namespace.Namespace == ns.Name {
+					for _, gvk := range ns.GVKs {
+						if gvk.Group == gvr.Group && gvk.Version == gvr.Version && gvk.Kind == gvr.Resource {
+							compatible = false
+						}
+					}
+				}
+			}
+			if !compatible {
+				continue
+			}
+
+			list, err := dynamic.Resource(gvr).Namespace(namespace.Namespace).List(metav1.ListOptions{})
+			if err != nil {
+				log.Trace(err)
+				return err
+			}
+
+			if len(list.Items) > 0 {
+				NamespaceResource := migapi.MigAnalyticNamespaceResource{
+					Group:   group,
+					Version: version,
+					Kind:    r.Name,
+					Count:   len(list.Items),
+				}
+				if !compatible {
+					namespace.IncompatibleK8SResources = append(namespace.IncompatibleK8SResources, NamespaceResource)
+					namespace.IncompatibleK8SResourceTotal += len(list.Items)
+				} else if excluded {
+					namespace.ExcludedK8SResources = append(namespace.ExcludedK8SResources, NamespaceResource)
+					namespace.ExcludedK8SResourceTotal += len(list.Items)
+				} else {
+					namespace.K8SResources = append(namespace.K8SResources, NamespaceResource)
+					namespace.K8SResourceTotal += len(list.Items)
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func (r *ReconcileMigAnalytic) analyzeImages(client k8sclient.Client, namespace *migapi.MigAnalyticNamespace, listImages bool, maxListImages int) error {
+	imageStreamList := imagev1.ImageStreamList{}
+
+	err := client.List(context.TODO(), k8sclient.InNamespace(namespace.Namespace), &imageStreamList)
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+
+	for _, im := range imageStreamList.Items {
+		internalRegistry, err := GetRegistryInfo(4, 4, client)
+		if err != nil {
+			log.Trace(err)
+			return err
+		}
+
+		for _, tag := range im.Status.Tags {
+			for i := len(tag.Items) - 1; i >= 0; i-- {
+				dockerImageReference := tag.Items[i].DockerImageReference
+				if len(internalRegistry) > 0 && strings.HasPrefix(dockerImageReference, internalRegistry) {
+					image := &imagev1.Image{}
+					err := client.Get(
+						context.TODO(),
+						types.NamespacedName{
+							Name: tag.Items[i].Image,
+						},
+						image)
+					if err != nil {
+						log.Trace(err)
+						return err
+					}
+
+					dockerImageMetadata := docker10.DockerImage{}
+					err = json.Unmarshal(image.DockerImageMetadata.Raw, &dockerImageMetadata)
+					if err != nil {
+						log.Trace(err)
+						return err
+					}
+
+					namespace.ImageCount += 1
+					namespace.ImageSizeTotal.Add(resource.MustParse((strconv.Itoa(int(dockerImageMetadata.Size)/MiB) + MiBSuffix)))
```

Comment:
I think we should try to decompose `analyzeImages()` into the major pieces bring down line lengths.